slider/curve.py

```diff
@@ -0,0 +1,179 @@
+from abc import ABCMeta, abstractmethod
+import math
+
+import numpy as np
+from scipy.misc import comb
+
+from .position import Position
+
+
+class Curve(metaclass=ABCMeta):
+    _kind_dispatch = {}
+
+    @classmethod
+    def from_kind_and_points(cls, kind, points):
+        try:
+            subcls = cls._kind_dispatch[kind]
+        except KeyError:
+            raise ValueError(f'unknown curve type: {kind!r}')
+
+        return subcls(points)
+
+    @abstractmethod
+    def __call__(self, t):
+        """Compute the position of the curve at time ``t``.
+
+        Parameters
+        ----------
+        t : float
+            The time along the distance of the curve in the range [0, 1]
+
+        Returns
+        -------
+        position : Position
+            The position of the curve.
+        """
+        raise NotImplementedError('__call__')
+
+    def __init_subclass__(cls):
+        cls._kind_dispatch
+        for meth in super(cls, cls).__abstractmethods__:
+            impl = getattr(cls, meth)
+            if impl.__doc__ is None:
+                impl.__doc__ = getattr(super(cls, cls), meth).__doc__
+
+        for kind in cls.kinds:
+            cls._kind_dispatch[kind] = cls
+
+
+class Bezier(Curve):
+    kinds = 'LB'
+
+    def __init__(self, points):
+        self.points = points
+        self._coordinates = np.array(points).T
+
+    def __call__(self, t):
+        points = self.points
+
+        n = len(points) - 1
+        ixs = np.arange(n + 1)
+        x, y = np.sum(
+            comb(n, ixs) *
+            (1 - t) ** (n - ixs) *
+            t ** ixs *
+            self._coordinates,
+            axis=1,
+        )
+        return Position(x, y)
+
+
+class Passthrough(Curve):
+    kinds = 'P'
+
+    def __init__(self, points):
+        self.points = points
+        self._center = center = get_center(*self.points)
+
+        coordinates = np.array(points) - center
+
+        # angles of first and last points
+        start_angle, end_angle = np.arctan2(coordinates[::2, 1], coordinates[::2, 0])
```

Comment:
I only wanted the first and last (I don't use the middle angle anymore) and couldn't think of a neater way to do it. Feel free to change to something more clear